.github/workflows/test_impl.yml

```diff
@@ -0,0 +1,132 @@
+name: Reusable build
+on:
+  workflow_call:
+    inputs:
+      runs_on:
+        description: Runner to run the job on
+        required: true
+        type: string
+
+      container:
+        description: "The container object as a JSON string (leave empty to run natively)"
+        required: true
+        type: string
+
+      conan_profile:
+        description: Conan profile to use
+        required: true
+        type: string
+
+      build_type:
+        description: Build type
+        required: true
+        type: string
+
+      unit_tests:
+        description: Whether to run unit tests
+        required: true
+        type: boolean
+
+      integration_tests:
+        description: Whether to run integration tests
+        required: true
+        type: boolean
+
+      compiler:
+        description: Compiler
+        required: true
+        type: string
+
+      sanitizer:
+        description: Sanitizer to use
+        required: false
+        type: string
+        default: "false"
+
+jobs:
+  test:
+    name: Test ${{ inputs.container != '' && 'in container' || 'natively' }}
+    runs-on: ${{ inputs.runs_on }}
+    container: ${{ inputs.container != '' && fromJson(inputs.container) || null }}
+
+    steps:
+      - name: Clean workdir
+        if: ${{ runner.os == 'macOS' }}
+        uses: kuznetsss/workspace-cleanup@80b9863b45562c148927c3d53621ef354e5ae7ce #v1.0
+
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: clio_tests_${{ runner.os }}_${{ inputs.build_type }}_${{ inputs.conan_profile }}
+
+      - name: Run clio_tests
+        run: |
+          chmod +x ./clio_tests
+          ${{ inputs.sanitizer != 'false' && './.github/scripts/execute-tests-under-sanitizer' || '' }} ./clio_tests
+
+      - name: Check for sanitizer report
+        shell: bash
+        id: check_report
+        run: |
+          if ls .sanitizer-report/* 1> /dev/null 2>&1; then
+            echo "found_report=true" >> $GITHUB_OUTPUT
+          else
+            echo "found_report=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Upload sanitizer report
+        if: ${{ steps.check_report.outputs.found_report == 'true' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ inputs.compiler }}_${{ inputs.sanitizer }}_report
+          path: .sanitizer-report/*
+          include-hidden-files: true
+
+      # TODO: enable when we have fixed all currently existing issues from sanitizers
+      - name: Create an issue
+        if: ${{ false && steps.check_report.outputs.found_report == 'true' }}
```

Comment:
I uncommented this code to make sure it stays correct (code in comments tends to become incorrect over time).
This way the only change we will need to enable issue creating is remove `false` here